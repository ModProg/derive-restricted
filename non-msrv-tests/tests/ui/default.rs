use std::marker::PhantomData;

use derive_where::DeriveWhere;

#[derive(DeriveWhere)]
#[derive_where(Debug)]
struct DefaultOnStruct<T>(#[derive_where(default)] PhantomData<T>);

#[derive(DeriveWhere)]
#[derive_where(Clone)]
enum DefaultWithoutTrait<T> {
	#[derive_where(default)]
	A(PhantomData<T>),
}

#[derive(DeriveWhere)]
#[derive_where(Default)]
enum MissingDefault<T> {
	A(PhantomData<T>),
}

#[derive(DeriveWhere)]
#[derive_where(Default)]
enum DuplicateDefaultSeparate<T> {
	#[derive_where(default)]
	A(PhantomData<T>),
	#[derive_where(default)]
	B(PhantomData<T>),
}

#[derive(DeriveWhere)]
#[derive_where(Default)]
enum DuplicateDefaultSame<T> {
	#[derive_where(default, default)]
	A(PhantomData<T>),
}

#[derive(DeriveWhere)]
#[derive_where(Default)]
enum DuplicateDefaultSameSeparate<T> {
	#[derive_where(default)]
	#[derive_where(default)]
	A(PhantomData<T>),
}

fn main() {}
