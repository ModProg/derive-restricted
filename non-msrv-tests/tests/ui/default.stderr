error: unknown option
 --> tests/ui/default.rs:7:42
  |
7 | struct DefaultOnStruct<T>(#[derive_where(default)] PhantomData<T>);
  |                                          ^^^^^^^

error: `default` is only supported if `Default` is being implemented
  --> tests/ui/default.rs:12:17
   |
12 |     #[derive_where(default)]
   |                    ^^^^^^^

error: required `default` option on a variant if `Default` is being implemented
  --> tests/ui/default.rs:17:1
   |
17 | / #[derive_where(Default)]
18 | | enum MissingDefault<T> {
19 | |     A(PhantomData<T>),
20 | | }
   | |_^

error: multiple `default` options in enum
  --> tests/ui/default.rs:27:17
   |
27 |     #[derive_where(default)]
   |                    ^^^^^^^

error: duplicate `default` option
  --> tests/ui/default.rs:34:26
   |
34 |     #[derive_where(default, default)]
   |                             ^^^^^^^

error: duplicate `default` option
  --> tests/ui/default.rs:42:17
   |
42 |     #[derive_where(default)]
   |                    ^^^^^^^
